docs/application/web/guides/data/file-system.md

```diff
@@ -108,149 +107,261 @@ To manage file storages:
+       console.log(error);
+   }
+   tizen.filesystem.deleteFile(fullPath, callback, errorCallback);
    ```
 
-   The `File` object (in [mobile](../../../../org.tizen.web.apireference/html/device_api/mobile/tizen/filesystem.html#File), [wearable](../../../../org.tizen.web.apireference/html/device_api/wearable/tizen/filesystem.html#File), and [TV](../../../../org.tizen.web.apireference/html/device_api/tv/tizen/filesystem.html#File) applications) is returned in the success event handler.
+## Retrieving Files and File Details
 
-2. To retrieve a list of all the files and their directories located in a specified directory, use the `listFiles()` method of the `File` object:`function onResolveSuccess(dir) {    dir.listFiles(onsuccess);}`The method returns an array of `File` objects.
+You can retrieve a list of files or file URIs using the `listDirectory()` and `toURI()` methods. The URI can be used to identify the file, for example, by using it as the `src` attribute on an HTML `img` element.
 
-3. To retrieve the file URI, use the `toURI()` method:
+1. To retrieve a list of all files and directories located in a specified directory, use the `listDirectory()` method of the `filesystem` object. The method returns an array of `DOMString` elements, each beeing a path with name of file inside a directory pointed by `path`.
 
    ```
    function onsuccess(files) {
        for (var i = 0; i < files.length; i++) {
-           /* Display the file name and URI */
-           console.log('File name is ' + files[i].name + ', URI is ' + files[i].toURI());
+           /* Display the file path with name */
+           console.log('File path and name is: ' + files[i]);
+       }
+   }
+   function onerror(error) {
+       console.log(error);
+   }
+   tizen.filesystem.listDirectory('documents/subDir', onsuccess, onerror);
    ```
 
-4. To retrieve the file content as a DOMString, use the `readAsText()` method.
-
-   The encoding input parameter of the method defines the format in which the file content is returned.
+2. To retrieve the file URI, use the `toURI()` method:
 
    ```
-           if (files[i].isDirectory == false) {
-               files[i].readAsText(function(str) {
-                   console.log('File content: ' + str);
-               }, null, 'UTF-8');
-           }
+   function onsuccess(files) {
+       for (var i = 0; i < files.length; i++) {
+           /* Display the file URI */
+           console.log('File URI is: ' + toURI(files[i]));
        }
    }
+   function onerror(error) {
+       console.log(error);
+   }
+   tizen.filesystem.listDirectory('documents/subDir', onsuccess, onerror);
    ```
 
+3. To retrieve the file content as a DOMString, binary data or to put it's contents to a `Blob` object use one of the following functions on `FileHandle` object opened in modes appropriate for reading ('r', 'rw').
+
+   - First open a file using `openFile()` function with mode allowing to read it's contents.
```

Comment:
Rephrase to:

Firstly, open a file using the `openFile()` function with mode allowing to read its contents.

The "function with mode" part is not clear. What mode?

Reply:
Done.